### PR TITLE
Use new iroh blobs methods for file operations

### LINKED ISSUE #42: Filesystem operation - Backend
Add
Remove
Get

### PR DESCRIPTION
**Pull request by tripledoublev**
Closes hyphacoop/openarchive-dweb-backend#42

### CODE REVIEW

File: src/repo.rs [hunk 1]
```diff
@@ -172,24 +185,48 @@ impl Repo {
...
-        let collection_name = self.get_name().await?;
-        
-        // Try to retrieve the collection hash
-        if let Ok(collection_hash) = self.iroh_blobs.collection_hash(&collection_name).await {
-            // If the collection exists, return the hash
-            return Ok(collection_hash);
+        // Try to get the collection hash from the DHT
+        if let Ok(collection_hash) = self.get_hash_from_dht().await {
+            // Try to retrieve the collection name from the hash
+            if let Ok(collection_name) = self.iroh_blobs.get_name_from_hash(&collection_hash).await {
```
Comment (RangerMauve) [inline src/repo.rs]: We shouldn't need to check the DHT, just check if the name exists in the iroh_blobs store

File: src/repo.rs [hunk 2]
```diff
@@ -199,46 +236,41 @@ impl Repo {
...
     pub async fn delete_file(&self, file_name: &str) -> Result<Hash> {
         self.check_write_permissions()?;
 
         // Ensure the collection exists before deleting a file
-        self.get_or_create_collection().await?;
+        let collection_hash = self.get_or_create_collection().await?;
 
-        let collection_name = self.get_name().await?;
+        // Get the collection name from the collection hash
+        let collection_name = self.iroh_blobs.get_name_from_hash(&collection_hash).await?;
```
Comment (RangerMauve) [inline src/repo.rs]: This should not be necessary, the name will always be the repo id

File: src/repo.rs [hunk 3]
```diff
@@ -253,27 +285,52 @@ impl Repo {
...
     pub async fn upload(&self, file_name: &str, data_to_upload: Vec<u8>) -> Result<Hash> {
         self.check_write_permissions()?;
 
         // Ensure the collection exists before uploading
-        self.get_or_create_collection().await?;
+        let collection_hash = self.get_or_create_collection().await?;
 
-        let collection_name = self.get_name().await?;
+        // Get the collection name from the collection hash if needed
+        let collection_name = self.iroh_blobs.get_name_from_hash(&collection_hash).await?;
```
Comment (RangerMauve) [inline src/repo.rs]: Again, we should use the repo name always

**Review by RangerMauve — CHANGES_REQUESTED** [review]

File: src/repo.rs (hunk 1 above)
Comment (tripledoublev) [reply src/repo.rs]: but if im not mistaken, we don't necessarily have the name. for example, if we get an url invitation to join a group.

Comment (tripledoublev) [reply src/repo.rs]: ok i get it now :)

Comment (tripledoublev) [reply src/repo.rs]: b38d2fde3edde400f42b0b0c7ea5a93fbe83124b

File: src/repo.rs (hunk 2 above)
Comment (tripledoublev) [reply src/repo.rs]: bbd6842ca77494ad362d4fd1be9bd03b531126a7

File: src/repo.rs (hunk 3 above)
Comment (tripledoublev) [reply src/repo.rs]: c15ec03d89f6ef8c93dad78a8ef64a756106d929

**Comment by RangerMauve** [comment]
Replaced by #16

**Pull request closed without merging**